### PR TITLE
Add socket arg to AsyncModbusTcpClient

### PR DESCRIPTION
**Pull request by dsblendo**
@janiversen Let me know if this is a reasonable request. I would like to pass an already created socket directly into `AsyncModbusTcpClient`, to pass it directly into `loop.create_connection()`.

### CODE REVIEW

File: pymodbus/transport/transport.py [hunk 1]
```diff
@@ -87,6 +88,7 @@ class CommParams:
     host: str = "localhost" # On some machines this will now be ::1
     port: int = 0
     source_address: tuple[str, int] | None = None
+    socket: socket.socket | None = None
```
Comment (janiversen) [inline pymodbus/transport/transport.py:91]: This worries me quite a bit, because comm_params was easy printable parameters.

File: pymodbus/transport/transport.py [hunk 2]
```diff
@@ -231,6 +233,7 @@ def init_setup_connect_listen(self, host: str, port: int) -> None:
                 self.handle_new_connection,
                 host,
                 port,
+                sock=self.comm_params.socket,
```
Comment (janiversen) [inline pymodbus/transport/transport.py:236]: This affects both the server (listen) and client (connect), but you have not changed the server part.

This really needs to be tested, what happens if you pass none.

**Review by janiversen — CHANGES_REQUESTED** [review]
I have to think more about this change, for a couple of reasons:
- how to handle reconnect (could reconnect to another server)
- how to interpret host/port parameters, socket comes connected
- How to handle the server (transport is shared)
- this needs new tests cases in transport to ensure there are no side effects.

With the questions properly solved I do not objections.

**Comment by dsblendo** [comment]
@janiversen thanks for the feedback. I will review the lift required to make these changes and meet your requirements to see if it is worthwhile.
